adapter/test/stdio/stdio_adapter_basic_test.cpp

```diff
@@ -0,0 +1,1293 @@
+TEST_CASE("Open", "[process=" + std::to_string(info.comm_size) +
+                      "]"
+                      "[operation=single_open]"
+                      "[repetition=1][file=1]") {
+  pretest();
+  SECTION("open non-existant file") {
+    FILE* fd = fopen(info.new_file.c_str(), "r");
+    REQUIRE(fd == nullptr);
+    fd = fopen(info.new_file.c_str(), "r+");
+    REQUIRE(fd == nullptr);
+  }
+
+  SECTION("truncate existing file and write-only") {
+    FILE* fd = fopen(info.existing_file.c_str(), "w");
+    REQUIRE(fd != nullptr);
+    int status = fclose(fd);
+    REQUIRE(status == 0);
+  }
+  SECTION("truncate existing file and read/write") {
+    FILE* fd = fopen(info.existing_file.c_str(), "w+");
+    REQUIRE(fd != nullptr);
+    int status = fclose(fd);
+    REQUIRE(status == 0);
+  }
+
+  SECTION("open existing file") {
+    FILE* fd = fopen(info.existing_file.c_str(), "r+");
+    REQUIRE(fd != nullptr);
+    int status = fclose(fd);
+    REQUIRE(status == 0);
+    fd = fopen(info.existing_file.c_str(), "r");
+    REQUIRE(fd != nullptr);
+    status = fclose(fd);
+    REQUIRE(status == 0);
+  }
+
+  SECTION("append write new file") {
+    FILE* fd = fopen(info.existing_file.c_str(), "a");
```

Comment:
The `SECTION` says "new file" but you're opening `existing_file`.

Reply:
fixed